### PR TITLE
Introduce reinitialise of database in pstress

### PR DESCRIPTION
**Pull request by mohitj1988**
Problem:
It is seen that sometimes due to server crashes, the datadir gets into an unrecoverable state
and server restart fails. This causes all the subsequent pstress trial to fail.

Solution:
Whenever a scenario like this will occur, pstress will wait for 2 server failures, after which
the data directory will be re-initialised. pstress will create fresh data and continue the subsequent
trials without failing.

### CODE REVIEW

**Review by rahulmalik87 — CHANGES_REQUESTED** [review]
Can you introduce new option to reinitialise the database.
something like 
--reintialize-probabality N. 
when 100. It will always reinitialise the datbase in each step
when  0. It will never reinitialise (unless crash recovery is failed)
between 0-100(N), It will initialise on any step based on N probability.
All add new option --same-seed, It will use same seed  in reinitialise else generate new seed to pass to pstress. 
Also part of testing it is also worth to modify some of mysqld options. 

basically these changes will allow us to do more testing in one pstress test.

**Comment by mohitj1988** [comment]
> alise else generate new seed to pass to pstress

IMO, we should keep things simpler, or else it will be a nightmare for a QA to reproduce the incidents/crashes. Even with same seed in multi-threaded scenarios, I am unable to repeat crashes. Adding reinitiasation will contribute to more chaos.

From a QA perspective, pstress is very effective at the moment without re-initialisation. With frequent re-initialsation, the DB state will be lost which will eventually lose crashing scenarios. We should restrict reinitialisation only for cases when pstress is unable to continue.

Secondly, the current driver script only adds configuration variables. It does not have the capacity of adding new options to the driver script.Adding random options like "--same-seed" will only confuse the user. Currently, pstress driver script generates a random number as seed and passes it to pstress workload script. Throughout the steps, the seed remains the same which is helpful in reproducing the crashes. Imagine a scenario where in every trial seed is changing and re-initiasation is happening. We will end up no-where near in repeating issues. Each trial will be like running pstress workload script independently.

All the above points are from a QA perspective. We are the actual users :)

**Comment by rahulmalik87** [comment]
> gs simpler, or else it will be a nightmare for a QA to reproduce the incidents/crashes. Even with same seed in multi-threaded scenario

It is feature request. 

> > alise else generate new seed to pass to pstress
> 
> IMO, we should keep things simpler, or else it will be a nightmare for a QA to reproduce the incidents/crashes. Even with same seed in multi-threaded scenarios, I am unable to repeat crashes. Adding reinitiasation will contribute to more chaos.
> 
> From a QA perspective, pstress is very effective at the moment without re-initialisation. With frequent re-initialsation, the DB state will be lost which will eventually lose crashing scenarios. We should restrict reinitialisation only for cases when pstress is unable to continue.
> 
> Secondly, the current driver script only adds configuration variables. It does not have the capacity of adding new options to the driver script.Adding random options like "--same-seed" will only confuse the user. Currently, pstress driver script generates a random number as seed and passes it to pstress workload script. Throughout the steps, the seed remains the same which is helpful in reproducing the crashes. Imagine a scenario where in every trial seed is changing and re-initiasation is happening. We will end up no-where near in repeating issues. Each trial will be like running pstress workload script independently.
> 
> All the above points are from a QA perspective. We are the actual users :)

@Mohit Joshi We are building a product. It needs features. I am not asking you to include --reintialize-probabality in your every test. you can set it 0. and do your testing the way the want. 
but this feature is important as it can help to do more testing one pstress test. 
It is important feature request. If you need some other opinion you can Satya or some other guy.

**Comment by rahulmalik87** [comment]
Or We can separate the request of reloading of pstress tables( through --prepare). I agree with you. lets handle just reinitialize of database as part of the bug fix

**Review by rahulmalik87 — APPROVED** [review]
LGTM.
